### PR TITLE
Convert AccountViewHolder from Java to Kotlin

### PR DESCRIPTION
**Pull request by nikclayton**
Use view binding in the converted code, which requires small changes in code that calls constructors.

Pass showBotOverlays as a parameter, rather than having the code reach in to the shared preferences, fixing a layering violation. This affects callers and classes derived from AccountAdapter.

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    fun setupWithAccount(
+        account: TimelineAccount,
+        animateAvatar: Boolean,
+        animateEmojis: Boolean,
+        showBotOverlay: Boolean
+    ) {
+        accountId = account.id
+
+        val format = binding.accountUsername.context.getString(R.string.post_username_format)
+        val formattedUsername = String.format(format, account.username)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt:27]: `getString` has overloads for passing in the formatting arg directly

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        )
+        binding.accountDisplayName.text = emojifiedName
+
+        val avatarRadius = binding.accountAvatar.context.resources
+            .getDimensionPixelSize(R.dimen.avatar_radius_48dp)
+        loadAvatar(account.avatar, binding.accountAvatar, avatarRadius, animateAvatar)
+
+        if (showBotOverlay && account.bot) {
+            binding.accountAvatarInset.visibility = View.VISIBLE
+            binding.accountAvatarInset.setImageResource(R.drawable.bot_badge)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt:43]: could set that in xml already and then just do
`binding.accountAvatarInset.visible(showBotOverlay && account.bot)`

Maybe even call `accountAvatarInset` `accountBotBadge`.

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.keylesspalace.tusky.entity.TimelineAccount
+import com.keylesspalace.tusky.interfaces.AccountActionListener
+import com.keylesspalace.tusky.interfaces.LinkListener
+import com.keylesspalace.tusky.util.emojify
+import com.keylesspalace.tusky.util.loadAvatar
+
+class AccountViewHolder(
+    private val binding: ItemAccountBinding
+) : RecyclerView.ViewHolder(binding.root) {
+    private var accountId: String? = null
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt:16]: `lateinit` to avoid `!!` later maybe?

**Review by connyduck — COMMENTED** [review]
nice improvement, thx

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt (hunk 1 above)
Comment (nikclayton) [reply app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt]: Done.

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt (hunk 2 above)
Comment (nikclayton) [reply app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt]: Done.

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt (hunk 3 above)
Comment (nikclayton) [reply app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt]: Done.

**Comment by nikclayton** [comment]
Thanks for the feedback, is does make the code a little simpler.

**Pull request merged**